drivers/pwm/pwm_nrf5_sw.c

```diff
@@ -160,31 +199,58 @@ static int pwm_nrf5_sw_pin_set(const struct device *dev, uint32_t pwm,
 	/* calc div, to scale down to fit in 16 bits */
 	div = period_cycles >> 16;
```

Comment:
RTC has 24-bit compare registers.
But this "scaling" is anyway incorrect, even for the 16-bit TIMER case. Instead, something like this should be done above (for RTC, the TIMER case can be corrected in a separate PR):
```
if (USE_RTC && period_cycles > BITMASK(24)) {
	LOG_ERR("Too long period (%u)", period_cycles);
	return -EINVAL;
}
```

Reply:
Am I right then, that this scaling (`div = period_cycles >> 16;`) should be dropped entirely for both RTC and TIMER?

Reply:
I think so. Currently it just uses incorrect settings, and thus gives incorrect PWM output. It will be better to return an error with some hint to the user, like in the mcux driver: https://github.com/zephyrproject-rtos/zephyr/blob/3c79b565fe7e6afed3687d95dd3e70bb93699a88/drivers/pwm/pwm_mcux.c#L58-L64

Reply:
Added similar check for both TIMER and RTC.